### ISSUE TITLE
Killing non related xmllint processes

### ISSUE THREAD

**Issue by pchris7489** [mluis7/xml2xpath#15]
There is a bug where `xml2xpath.sh` will kill non related `xmllint` processes.

The line `pkill -f xmllint` doesn't discriminate on the xmllint process the script opened itself. This can crash other instances of `xml2xpath.sh` or crash other non related scripts.

In the function `get_xml_tree`, there should be a capturing of the PID of xmllint so it can be later be tracked and destroyed in the trap function. For this part, also :

```
        if [ "$isHtml" -eq 1 ]; then
            print_all_xpath | "${lint_cmd[@]}" "$xml_file" | "${dbg_cmd[@]}"
        else
            (set_root_ns; print_all_xpath) | "${lint_cmd[@]}" "$xml_file" | "${dbg_cmd[@]}"
        fi
```

**Comment by mluis7** [comment]
@pchris7489 
Thanks for taking the time to report the errors.
The fix to this issue is to pass the parent PID to `pkill` as `pkill --parent $$ -f xmllint` so only child processes of current script will be terminated.
I suspect the parallel processing reported in the other issue is related to this one. Capturing stderr could show something like
`Timeout reading from file descriptor $1 $2 . Current timeout: $rtout secs. Try extending the timeout with: XML_XPATH_RTOUT=[int or float > $rtout] xml2xpath.sh ...`
caused by the defunct `xmllint` process.

**Comment by pchris7489** [comment]
Just tried it on my main machine (I am using this script on different platforms, Ubuntu, Android and Windows), and the command `pkill --parent $$ -f xmllint` is working on Cygwin (which is very promissing, because Cygwin is a port). Also, indeed, this is the type of error message that is displayed, because the xmllint for the other processes are killed
Thank you, I will more heavily try this solution on a more heavy workload.

**Comment by mluis7** [comment]
Fixed and new version 0.12.0 available.
